### DISCUSSION TITLE
Failed to start gauge API: Timed out connecting to java

### DISCUSSION THREAD

**Discussion by suman-gs** [getgauge/gauge#2162] [General]
I am geeting API timed out error when i added more gradle dependencies ,attached sample project for reference

Steps to follow: 
1) Build the grdale dependecies
2) Run   --> gauge run specs

then getting below error:
![image](https://user-images.githubusercontent.com/92427469/141747923-07dd994c-5001-41de-96ee-5c115a60179e.png)

Attaching sample project
[gaugeTest_2.zip](https://github.com/getgauge/gauge/files/7537061/gaugeTest_2.zip)

Gauge.log:

[gauge.log](https://github.com/getgauge/gauge/files/7537084/gauge.log)
PLease let me know what configurations do i need to change here.

**Comment by sriv** [comment]
can you try running `./gradlew gauge` instead of `gauge run specs`? This works for me.

**Reply by suman-gs** [reply]
HI Srikanth, 
"./gradlew gauge.."  is working fine , but our requirement is to run with "gauge run specs"

if you replace with  below build.gradle it will work
````
plugins {
    id 'java'
    id 'org.gauge' version '1.8.2'
}

group 'org.example'
version '1.0-SNAPSHOT'

sourceCompatibility = 11
targetCompatibility = 11

repositories {
    mavenCentral()
}

dependencies {
    implementation 'com.thoughtworks.gauge:gauge-java:0.7.15'
    implementation 'io.rest-assured:rest-assured:4.4.0'
    implementation 'io.rest-assured:json-path:4.4.0'
    implementation 'io.rest-assured:json-schema-validator:4.4.0'
    testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.20.2'
}

jar {
    zip64=true
    from {
        configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
    }
}
````
![image](https://user-images.githubusercontent.com/92427469/142036217-282f3d73-9ddf-4205-9cd8-6083dc26ef75.png)

**Reply by sriv** [reply]
>  "./gradlew gauge.." is working fine , but our requirement is to run with "gauge run specs"

Would like to know more about this, it is normally recommended to let gradle orchestrate the build for when you use it in a project.
This way gradle can figure out the right classpath for the project and set it before launching gauge.

The gauge-java launcher tries to get set the classpath, but you should think of a best effort attempt rather than being foolproof.

In your project, `gauge run specs` leads to `RestAssured` not getting resolved properly.
